src/lib/api/ext/img-srcset.spec.ts

```diff
@@ -0,0 +1,353 @@
+    });
+    expect(nodes[2].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['xs'][0]}`,
+      media: breakpoints.findByAlias('xs').mediaQuery
+    });
+  });
+
+  it('should update source elements srcset values when srcset input properties change', () => {
+    const template = `
+      <picture>
+          <img  style="width:auto;"
+                src="${DEFAULT_SRC}"
+                [srcset.xs]="xsSrcSet"
+                [srcset.lg]="lgSrcSet"
+                [srcset.md]="mdSrcSet" >
+      </picture>
+    `;
+    componentWithTemplate(template);
+    fixture.detectChanges();
+
+    fixture.componentInstance.xsSrcSet = SRCSET_URLS_MAP['xs'][1];
+    fixture.componentInstance.mdSrcSet = SRCSET_URLS_MAP['md'][1];
+    fixture.componentInstance.lgSrcSet = SRCSET_URLS_MAP['lg'][1];
+    fixture.detectChanges();
+
+    let nodes = queryFor(fixture, 'source');
+
+    expect(nodes.length).toBe(3);
+    expect(nodes[0].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['lg'][1]}`,
+      media: breakpoints.findByAlias('lg').mediaQuery
+    });
+    expect(nodes[1].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['md'][1]}`,
+      media: breakpoints.findByAlias('md').mediaQuery
+    });
+    expect(nodes[2].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['xs'][1]}`,
+      media: breakpoints.findByAlias('xs').mediaQuery
+    });
+  });
+
+  it('should work with overlapping breakpoints', () => {
+    const template = `
+      <picture>
+          <img  style="width:auto;"
+                src="${DEFAULT_SRC}"
+                srcset.lt-xl="${SRCSET_URLS_MAP['lt-xl'][0]}"
+                srcset.xs="${SRCSET_URLS_MAP['xs'][0]}"
+                srcset.lt-lg="${SRCSET_URLS_MAP['lt-lg'][0]}" >
+      </picture>
+    `;
+    componentWithTemplate(template);
+    fixture.detectChanges();
+
+    let nodes = queryFor(fixture, 'source');
+    expect(nodes[0].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['lt-xl'][0]}`,
+      media: breakpoints.findByAlias('lt-xl').mediaQuery
+    });
+    expect(nodes[1].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['lt-lg'][0]}`,
+      media: breakpoints.findByAlias('lt-lg').mediaQuery
+    });
+    expect(nodes[2].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['xs'][0]}`,
+      media: breakpoints.findByAlias('xs').mediaQuery
+    });
+  });
+
+  describe('with responsive api', () => {
+
+    it('should work with a isolated image element and responsive srcset(s)', () => {
+      componentWithTemplate(`
+        <img [srcset]="xsSrcSet"
+             [srcset.md]="mdSrcSet">
+      `);
+
+      let img = queryFor(fixture, 'img')[0].nativeElement;
+
+      matchMedia.activate('md');
+      fixture.detectChanges();
+      expect(img).toBeDefined();
+      expect(img).toHaveAttributes({
+        src: fixture.componentInstance.mdSrcSet
```

Comment:
@ThomasBurleson  With this test, we check that the src attribute is updated. However, srcset is still in the dom (with the old value corresponding to xsSrcSet). In case srcset value uses the pixel density 1x, it will have the priority, and we will not have the intended behaviour. I think that the test should be updated to do validation on both src and srcset. But most importantly, what should we do? Update in this case the srcset value and not introduce src? Or remove the srcset from the dom ?